core/event_api.php

```diff
@@ -68,6 +68,17 @@ function event_declare_many( array $p_events ) {
 	}
 }
 
+/**
+ * Determine if a given event has been declared
+ * @param string $p_name Event name
+ * @return bool True if the even has been declared
+ */
+function event_is_declared( $p_name ) {
+	global $g_event_cache;
+
+	return array_key_exists( $p_name, $g_event_cache );
```

Comment:
Other functions use `isset` instead of `array_key_exists` to check if an event is declared.
e.g.

```
	if( !isset( $g_event_cache[$p_name] ) ) {
		error_parameters( $p_name );
		trigger_error( ERROR_EVENT_UNDECLARED, WARNING );
		return null;
	}
```
Worth to consider also the difference concerning `null` value treatment, e.g. check [this](https://coderwall.com/p/q9erfw/isset-vs-array_key_exists).

Reply:
> Other functions use isset instead of array_key_exists 

right, all the items in `$g_event_cache` are internally built and null is not an expected value.

Reply:
I generally use array_key_exists() by default to be on the safe side, but considering @cproensa's comment I'll switch to isset.